ql/src/test/org/apache/hadoop/hive/ql/optimizer/TestSharedWorkOptimizer.java

```diff
@@ -323,39 +314,72 @@ public void testMapJoinCacheReuse() {
     Operator<?> rsC2 = OperatorFactory.getAndMakeChild(getReduceSinkDesc(), smallTableC);
 
     // case 1. MapJoin1: (big, A, B), MapJoin2: (big, A, B)
-    List<Operator<?>> joinSource1 = Arrays.asList(ts1, rsA1, rsB1);
-    List<Operator<?>> joinSource2 = Arrays.asList(ts2, rsA2, rsB2);
+    setupJoinOperatorsAndTest(ts1, ts2, rsA1, rsA2, rsB1, rsB2, 0, 0,
+        JoinType.INNER.ordinal(), JoinType.INNER.ordinal(), true);
 
-    MapJoinOperator mapJoin1 = (MapJoinOperator) OperatorFactory.getAndMakeChild(
-        cCtx, getMapJoinDesc(0), joinSource1);
-    MapJoinOperator mapJoin2 = (MapJoinOperator) OperatorFactory.getAndMakeChild(
-        cCtx, getMapJoinDesc(0), joinSource2);
+    // case 2. MapJoin1: (big, A, B), MapJoin2: (big, A, C)
+    setupJoinOperatorsAndTest(ts1, ts2, rsA1, rsA2, rsB1, rsC2, 0, 0,
+        JoinType.INNER.ordinal(), JoinType.INNER.ordinal(), false);
```

Comment:
It would be better to move each case to its test method because in the current form the context (ts1, smallTableA, rsA1, etc) of the tests is reused. The code or any test may makes changes on these objects and causes interference between test cases.

Reply:
Addressed this